xdsl/dialects/builtin.py

```diff
@@ -825,6 +821,17 @@ def __init__(
                 type = Float128Type()
             else:
                 raise ValueError(f"Invalid bitwidth: {type}")
+
+        value: float = data.data if isinstance(data, FloatData) else data
+        try:
+            # constrain value to precision of floating point type
+            value = type.unpack(type.pack([value]), 1)[0]
+        except NotImplementedError:
+            # allow full precision
+            pass
```

Comment:
This is a little scary, what do you think about instead manually check for the supported values here?